### PR TITLE
adds segmentStore Get, List, Delete

### PR DESCRIPTION
**Pull request by navillasa**
Get:
- requests pointer from pointerdb
- requests associated node IPs from overlay
- requests ranger from ecclient

Delete:
- requests pointer from pointerdb
- uses ecclient to request piece deletes
- deletes pointer from pointerdb

List:
- requests a page of pointers from pointerdb

Other notes:
* all API keys given to PointerDB are nil
* List function takes startingPath and endingPath as arguments, even though the PointerDB List function currently requires a StartingPath and Limit. So SegmentStore List will have to be updated when PointerDB List is updated.
* the function in overlayHelper needs to run parallel (and needs to be done by pointerdb)

---
This change is [Reviewable](https://reviewable.io/reviews/storj/storj/143)

### CODE REVIEW

File: pkg/segment/segment.go [hunk 1]
```diff
@@ -105,29 +108,124 @@ func (s *segmentStore) Put(ctx context.Context, path paths.Path, data io.Reader,
...
 
-// Get retrieves a file from the erasure code client with help from overlay and pointerdb
+// Get retrieves a file using erasure code, overlay, and pointerdb clients
 func (s *segmentStore) Get(ctx context.Context, path paths.Path) (ranger.Ranger, Meta, error) {
-	m := Meta{
-		Data: []byte(""),
+	m := Meta{}
+	// TODO: remove this chunk after pointerdb client interface merged
+	gr := &ppb.GetRequest{
+		Path:   []byte(fmt.Sprintf("%s", path)),
```
Comment (jtolds) [inline pkg/segment/segment.go]: nit: `fmt.Sprintf("%s", thing)` just calls `thing.String()`. so we can avoid fmt.Sprintf entirely?

File: pkg/segment/segment.go [hunk 2]
```diff
@@ -105,29 +108,124 @@ func (s *segmentStore) Put(ctx context.Context, path paths.Path, data io.Reader,
...
+
+	if pointer.Type != ppb.Pointer_REMOTE {
+		zap.L().Error("TODO: only getting remote pointers supported")
+		return nil, m, err
+	}
+
+	remoteSeg := pointer.Remote
+	var nodes []*opb.Node
+	for i := 0; i < len(remoteSeg.RemotePieces); i++ {
+		overlayRes, err := s.oc.Lookup(ctx, &opb.LookupRequest{NodeID: remoteSeg.RemotePieces[i].NodeId})
```
Comment (jtolds) [inline pkg/segment/segment.go]: shoot, would be nice to do this in parallel

even better would be if the pointerdb response had gotten this information from overlaycache for us.

no need to fix for now

File: pkg/segment/segment.go [hunk 3]
```diff
@@ -105,29 +108,124 @@ func (s *segmentStore) Put(ctx context.Context, path paths.Path, data io.Reader,
...
+		nodes = append(nodes, overlayRes.Node)
 	}
-	return nil, m, nil
+	pid := client.PieceID(remoteSeg.PieceId)
+	ecRes, err := s.ec.Get(ctx, nodes, s.rs, pid, pointer.Size)
+	if err != nil {
+		return nil, m, err
+	}
+
+	m.Data = pointer.Metadata
```
Comment (jtolds) [inline pkg/segment/segment.go]: is this the only m field to fill in?

File: pkg/segment/segment.go [hunk 4]
```diff
@@ -105,29 +108,124 @@ func (s *segmentStore) Put(ctx context.Context, path paths.Path, data io.Reader,
...
+
+	// TODO: remove this chunk after pointerdb client interface merged
+	pointer := &ppb.Pointer{}
+	err = proto.Unmarshal(pdbRes.Pointer, pointer)
+	if err != nil {
+		return err
+	}
+
+	// piece store client sends delete request
+	_, err = s.ps.Delete(ctx, &pspb.PieceDelete{Id: pointer.Remote.PieceId})
```
Comment (jtolds) [inline pkg/segment/segment.go]: oh, close! which piece store is this?

we should probably add delete to the ecclient, so it knows how to farm a delete out to all 40 piece store nodes

File: pkg/segment/segment.go [hunk 5]
```diff
@@ -105,29 +108,124 @@ func (s *segmentStore) Put(ctx context.Context, path paths.Path, data io.Reader,
...
+	}
+
+	res, err := s.pdb.List(ctx, lr)
+	if err != nil {
+		return nil, false, err
+	}
+
+	for _, path := range res.Paths {
+		var pathType []string
+		pathType = append(pathType, string(path[:]))
```
Comment (jtolds) [inline pkg/segment/segment.go]: you should use the path.NewPath constructor or whatever i think

**Comment by kaloyan-raev** [comment]
@navillasa There was a [discussion in Slack](https://storj.slack.com/archives/CAQV0AE2Z/p1530282568000501) to put all Store interfaces and implementations under /pkg/storage. Perhaps, it's a good moments to move the Segment Store under `/pkg/storage/segments`.

**Pull request merged**